jiant/tasks/tasks.py

```diff
@@ -3731,3 +3732,132 @@ def get_metrics(self, reset=False):
+class SentenceOrderTask(Task):
+    """ Task class for Sentence Order Prediction (SOP). See the ALBERT paper for details on SOP:
+        https://arxiv.org/abs/1909.11942.
+        We are currently using an unpreprocessed version of the Wikipedia corpus
+        (more specifically, the Wikidump data) that consists of 5% of the data. You can generate
+        the data by following the instructions from jiant/scripts/mlm.
+        There are several key things to note about our SOP ALBERT implementation.
+            1. We do not load the pretrained weights for the SOP head beacuse they are
+            unavailable in Huggingface. We only use the pretrained weights of the linear layer
+            from ALBERT that creates the pooled output used in SOP.
+            2. jiant supports a multitask version of the ALBERT pretraining regime, which requires
+            at each training step for each piece of text running that text through MLM and SOP head
+            at the same time.
+    """
+
+    def __init__(self, path, max_seq_len, name, **kw):
+        super(SentenceOrderTask, self).__init__(name, **kw)
+        self.path = path
+        self.max_seq_len = max_seq_len
+        self.n_classes = 2
+        self.train_data_text = None
+        self.val_data_text = None
+        self.test_data_text = None
+        self.files_by_split = {
+            "train": os.path.join(path, "train.txt"),
+            "val": os.path.join(path, "valid.txt"),
+            "test": os.path.join(path, "test.txt"),
+        }
+        self.val_metric_decreases = False
+        self.val_metric = "%s_accuracy" % self.name
+        self.acc_scorer = BooleanAccuracy()
+
+    def get_metrics(self, reset=False):
+        """Get metrics specific to the task"""
+        acc = self.acc_scorer.get_metric(reset)
+        return {"accuracy": acc}
+
+    def update_metrics(self, out, batch):
+        logits = out["logits"]
+        labels = out["labels"]
+        _, preds = logits.max(dim=1)
+        self.acc_scorer(preds, labels)
+
+    def get_data_iter(self, path):
+        """Loading data file and tokenizing the text
+        Args:
+            path: (str) data file path
+        """
+        f = open(path, "r")
+        text = [line.strip() for line in f]
+        for i in range(len(text) - 1):
+            if random.uniform(0, 1) > 0.5:
+                is_right_order = 1
+                sent_a = text[i]
+                sent_b = text[i + 1]
+            else:
+                is_right_order = 0
+                sent_a = text[i + 1]
+                sent_b = text[i]
+            sent_a_processed = tokenize_and_truncate(
+                self._tokenizer_name, sent_a, self.max_seq_len // 2
+            )
+            sent_b_processed = tokenize_and_truncate(
+                self._tokenizer_name, sent_b, self.max_seq_len // 2
+            )
+            yield (sent_a_processed, sent_b_processed, is_right_order)
+
+    def load_data(self):
+        pass
+
+    def process_split(
```

Comment:
The methods below still need a look from someone else.

Reply:
@zphang mind taking a look?

Reply:
Looks ok to me now. Jason left a few comments and I've fixed it according to his comments.